packages/blitz-next/src/index-server.ts

```diff
@@ -187,8 +187,8 @@ export const setupBlitzServer = ({plugins, onError}: SetupBlitzOptions) => {
     ): NextApiHandler<TResult | void> =>
     async (req, res) => {
       try {
-        await handleRequestWithMiddleware(req, res, middlewares)
-        return handler(req, res, res.blitzCtx)
+        middlewares.push((req, res) => handler(req, res, res.blitzCtx));
+        return await handleRequestWithMiddleware(req, res, middlewares);
```

Comment:
@jan-wilhelm The CI is failing because there are some type errors during the build process.